src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/seed/SeedJob.kt

```diff
@@ -0,0 +1,18 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.seed
+
+import java.util.UUID
+
+abstract class SeedJob <RowType> (
+  val id: UUID = UUID.randomUUID(),
+  val fileName: String,
+  val requiredColumns: Int
+) {
+  init {
+    if (fileName.contains("/") || fileName.contains("\\") || fileName.contains(".")) {
```

Comment:
Did you consider using a filesystem function to verify whether a file of the given name (and expected mime type) exists in the expected seed directory? Perhaps at this point of accepting or rejecting the async job it's not appropriate / possible  to take any external action?

Reply:
I had considered doing the does-file-exist check before returning the 202 but opted against it since we'd then have a situation where some failures are reported synchronously and some aren't (e.g. if the file then happens to be malformed, you'd have to check the logs.)   Can add this anyway though if you feel the pros outweigh the cons?

Reply:
right, that makes sense -- I thought it might be in consideration of the async processes 🚀